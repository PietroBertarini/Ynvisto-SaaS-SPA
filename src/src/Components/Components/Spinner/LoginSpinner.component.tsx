import React from 'react';

import { SpinnerContainer } from './spinner.styles';

/**
 * Basic Spinner component that is render when something is loading.
 */
const LoginSpinner = () => (
  <SpinnerContainer />
);

export default LoginSpinner;
