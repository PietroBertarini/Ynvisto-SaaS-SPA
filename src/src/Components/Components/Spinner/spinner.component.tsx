import React from 'react';

import { SpinnerContainer, SpinnerOverlay } from './spinner.styles';

/**
 * Basic Spinner component that is render when something is loading.
 */
const Spinner = () => (
  <SpinnerOverlay>
    <SpinnerContainer />
  </SpinnerOverlay>
);

export default Spinner;
