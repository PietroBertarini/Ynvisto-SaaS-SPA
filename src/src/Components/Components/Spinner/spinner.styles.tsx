import styled from 'styled-components';

export const SpinnerOverlay = styled.div`
  height: 70vh;
  width: 100%;
  display: flex;
  margin-left: 100px;
  justify-content: center;
  align-items: center;
`;

interface SpinnerProps {
  size?: number,
  spinnerColor?: string
}

// https://styled-components.com/docs/basics Props doc
export const SpinnerContainer = styled.div<SpinnerProps>`
  display: inline-block;
  width: ${(props) => (props.size ? props.size : 50)}px;
  height: ${(props) => (props.size ? props.size : 50)}px;
  border: 3px solid rgba(195, 195, 195, 0.6);
  border-radius: 50%;
  border-top-color: ${(props) => (props.spinnerColor ? props.spinnerColor : '#636767')};
  animation: spin 1s ease-in-out infinite;
  -webkit-animation: spin 1s ease-in-out infinite;
  @keyframes spin {
    to {
      -webkit-transform: rotate(360deg);
    }
  }
  @-webkit-keyframes spin {
    to {
      -webkit-transform: rotate(360deg);
    }
  }
`;
