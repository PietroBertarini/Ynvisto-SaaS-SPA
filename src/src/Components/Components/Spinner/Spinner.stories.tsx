import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Spinner from './spinner.component';

export default {
  title: 'Components/Spinner',
  component: Spinner,
} as Meta;

const Template: Story<{ }> = (args) => (<Spinner />
);

export const Full = Template.bind({});
