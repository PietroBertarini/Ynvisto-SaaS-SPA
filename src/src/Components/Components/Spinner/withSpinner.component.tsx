import React from 'react';
import Spinner from './spinner.component';

// eslint-disable-next-line react/prop-types
const WithSpinner = (WrappedComponent) => ({ isLoading, ...otherProps }) => (isLoading
  ? <Spinner /> : <WrappedComponent {...otherProps} />);

export default WithSpinner;
