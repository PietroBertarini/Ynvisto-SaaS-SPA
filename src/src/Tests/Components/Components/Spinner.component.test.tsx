import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '../../../Components/Components/Spinner/spinner.component';

test('Renders Spinner component and match to previously snapshot', () => {
  const component = shallow(
    <Spinner />,
  );
  expect(component).toMatchSnapshot();
});
