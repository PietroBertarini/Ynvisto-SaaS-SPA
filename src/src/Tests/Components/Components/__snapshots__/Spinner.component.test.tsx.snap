// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Spinner component and match to previously snapshot 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
